packages/block-library/src/site-title/edit.js

```diff
@@ -36,4 +39,19 @@ export default function SiteTitleEdit() {
 			/>
 		</>
 	);
+
+	const viewMode = (
+		<h1>{ title }</h1>
+	);
+
+	return ( props.canUpdate && props.isSelected ) ? editMode : viewMode;
```

Comment:
Can we disable the input instead of introducing this split rendering path?

It seems innocuous in this context, but split edit/view paths will make it really hard to keep a consistent rendering style, so it will be better if we enforce using a single rendering path from the start.

Reply:
+1 to thinking more about how split render paths can be avoided.

Reply:
I agree. We explored a read only mode for `RichText` a while ago that could be enabled based on the context (a locked template or block for example). Additionally it would be good to disable attribute updating.